controllers/user.js

```diff
@@ -189,8 +191,31 @@ const deleteUser = async (req, res) => {
 
         for (let i = 0; i < user.workouts.length; i++) {
             const workout = await Workout.findById(user.workouts[i]);
+
             if (workout.user.toString() === userID) {
-                await workout.deleteOne();
+                workout.user = null;
+
+                if(workout.company.length !==0 && !workout.endTime)
+                {      
+                    
+                    workout.user = workout.company[0];
+                    const newUser = await User.findById(workout.user);
+
+                    if(!newUser.workouts)
```

Comment:
There is no need to add workout to the user, he already is a part of the workout so the workout id is in his workouts array

Reply:
i just realized that it was implemented already, you just created your branch before the changes were made